src/help/pages/help-section/help-section.ts

```diff
@@ -1,23 +1,235 @@
+import { Page } from 'ionic-angular/navigation/nav-util';
+
 import { HelpWaitingRoomToCarPage } from './../help-waiting-room-to-car/help-waiting-room-to-car';
 import { HelpTestReportPage } from './../help-test-report/help-test-report';
 import { HelpJournalPage } from './../help-journal/help-journal';
 import { HelpFinalisationSubmissionPage } from './../help-finalisation-submission/help-finalisation-submission';
 import { HelpDebriefPage } from './../help-debrief/help-debrief';
-import { Component } from '@angular/core';
-import { NavController, NavParams } from 'ionic-angular';
-import { Page } from 'ionic-angular/navigation/nav-util';
+
+enum visibilityType {
+  collapsed = 'collapsed',
+  expanded = 'expanded'
+}
+
+interface IVisibility {
+  journal: visibilityType;
+  waitingRoomToCar: visibilityType;
+  inCarTesting: visibilityType;
+  debreif: visibilityType;
+  finalisationAndSubmission: visibilityType;
+}
 
 @Component({
   selector: 'page-help-section',
   templateUrl: 'help-section.html'
 })
 export class HelpSectionPage {
-  title: string = 'Help section';
+  title: string = 'Help guide';
+
   helpDebriefPage: Page = HelpDebriefPage;
   helpFinalisationSubmissionPage: Page = HelpFinalisationSubmissionPage;
   helpJournalPage: Page = HelpJournalPage;
   helpTestReportPage: Page = HelpTestReportPage;
   helpWaitingRoomToCarPage: Page = HelpWaitingRoomToCarPage;
 
+  visibility: IVisibility = {
+    journal: visibilityType.expanded,
+    waitingRoomToCar: visibilityType.collapsed,
+    inCarTesting: visibilityType.collapsed,
+    debreif: visibilityType.collapsed,
+    finalisationAndSubmission: visibilityType.collapsed
+  };
+
   constructor(public navCtrl: NavController, public navParams: NavParams) {}
+
+  // Expanding and collapsing all sections
+
+  isAtLeastOneCollapsed() {
+    return Object.keys(this.visibility).some(
+      (section) => this.visibility[section] === visibilityType.collapsed
+    );
+  }
+
+  isAllExpanded() {
+    return Object.keys(this.visibility).every(
+      (section) => this.visibility[section] === visibilityType.expanded
+    );
+  }
+
+  expandAll() {
+    this.visibility = {
+      journal: visibilityType.expanded,
+      waitingRoomToCar: visibilityType.expanded,
+      inCarTesting: visibilityType.expanded,
+      debreif: visibilityType.expanded,
+      finalisationAndSubmission: visibilityType.expanded
+    };
+  }
+
+  collapseAll() {
+    this.visibility = {
+      journal: visibilityType.collapsed,
+      waitingRoomToCar: visibilityType.collapsed,
+      inCarTesting: visibilityType.collapsed,
+      debreif: visibilityType.collapsed,
+      finalisationAndSubmission: visibilityType.collapsed
+    };
+  }
+
+  // Expanding and collapsing Journal section
+
+  isJournalVisibilityExpended() {
+    return this.visibility.journal === visibilityType.expanded;
+  }
+
+  isJournalVisibilityCollapsed() {
+    return this.visibility.journal === visibilityType.collapsed;
+  }
+
```

Comment:
Why do we need 2 methods for a toggle with only 2 options shouldn't we be just  isJournalVisibilityExpended() and !isJournalVisibilityExpended() to show and hide.

To me this seems to be over engineered where a boolean for each toggle would have been enough.

Reply:
Yes, you are right that for this specific example (if it never going to change) using a boolean would have been a solution. However I didn't choose that solution because in the majority of the cases when you start toggling boolean to represent the state of your application with that type is going to end up having bugs. For instant you, as a reader don't know that `false` means expanded or collapsed, but with enum is clear.

So this is a different way of thinking which is helping me to have a pure business logic around this list.

Reply:
I like the use of enums to enrich primitive values with more meaning and to increase readability. I think we should start using this style more often.